### PR TITLE
fix: Display the correct cursor when a reorder/resize operation is ongoing

### PR DESCRIPTION
**Pull request by timogasda**
### Description
There's currently a bug that prevents the correct mouse cursor to be displayed when a reorder or resize operation is ongoing. This is because the drag-and-drop operation sets `pointer-events: none` on the board item, which means that any `cursor` styles are ignored on the children. Instead, we're now applying the appropriate `cursor` on the Board itself.

Related links, issue #, if available: AWSUI-20820

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/board/__tests__/board.test.tsx [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+import boardStyles from "../../../lib/components/board/styles.css.js";
 import BoardItem from "../../../lib/components/board-item";
+import { mockController } from "../../../lib/components/internal/dnd-controller/__mocks__/controller";
+import { DragAndDropData } from "../../../lib/components/internal/dnd-controller/controller";
+import { Coordinates } from "../../../lib/components/internal/utils/coordinates";
 import createWrapper from "../../../lib/components/test-utils/dom";
 
+afterEach(cleanup);
+
+vi.mock("../../../lib/components/internal/dnd-controller/controller");
```
Comment (just-boris) [inline src/board/__tests__/board.test.tsx:16]: Other tests in this file relied on genuine instance of the controller. Can we avoid this mock and use mousedown/mouseup events instead?

(probably, it should be a public test-util too, but let's do manual events right here for now)

File: src/board/internal.tsx [hunk 2]
```diff
@@ -221,7 +229,12 @@ export function InternalBoard<D>({
         onActivateItem={focusItem}
       />
 
-      <div ref={containerRef} className={clsx(styles.root, { [styles.empty]: rows === 0 })}>
+      <div
+        ref={containerRef}
+        className={clsx(styles.root, currentOperation && styles[`current-operation-${currentOperation}`], {
```
Comment (just-boris) [inline src/board/internal.tsx:234]: User may drag the widget outside the board area. I think, we should apply the new class to the entire document to handle this

Comment (timogasda) [reply src/board/internal.tsx:234]: I was debating this too... I'll change it to apply on the body instead then.

File: src/board/__tests__/board.test.tsx (hunk 1 above)
Comment (timogasda) [reply src/board/__tests__/board.test.tsx:16]: I was struggling to get this to work with `fireEvent` or `.click()`. It wasn't updating my rendered component correctly. Any idea what I might be doing wrong?

File: src/board/__tests__/board.test.tsx [hunk 3]
```diff
@@ -99,4 +106,62 @@ describe("Board", () => {
...
+    );
+
+    const reorderClass = boardStyles["current-operation-reorder"];
+    expect(container.ownerDocument.body).not.toHaveClass(reorderClass);
+
+    const handle = createWrapper().findBoardItem()!.findDragHandle()!.getElement();
+    const user = userEvent.setup({ pointerEventsCheck: PointerEventsCheckLevel.Never });
+
+    // Start a reorder operation and check that the class is set
+    await user.pointer({ keys: "[MouseLeft>]", target: handle });
```
Comment (just-boris) [inline src/board/__tests__/board.test.tsx:131]: Why did you need this library here? What did not work when dispatching mousedown directly?

If we have a plan of converting this code into test-util, it will be more difficult when using an external dependency

File: src/board/internal.tsx [hunk 4]
```diff
@@ -167,6 +171,8 @@ export function InternalBoard<D>({
   useDragSubscription("discard", () => {
     dispatch({ type: "discard" });
 
+    document?.body?.classList.remove(styles["current-operation-reorder"], styles["current-operation-resize"]);
```
Comment (just-boris) [inline src/board/internal.tsx:174]: What is the condition when document or body is undefined?

File: src/internal/dnd-controller/__mocks__/controller.ts [hunk 5]
```diff
@@ -41,3 +41,5 @@ export const mockDraggable = {
 export function useDraggable() {
   return mockDraggable;
 }
+
+export const useDroppable = vi.fn() as any;
```
Comment (just-boris) [inline src/internal/dnd-controller/__mocks__/controller.ts:45]: Is this line needed in the latest revision?

Comment (timogasda) [reply src/internal/dnd-controller/__mocks__/controller.ts]: nope!

File: src/board/__tests__/board.test.tsx (hunk 3 above)
Comment (timogasda) [reply src/board/__tests__/board.test.tsx:131]: There were a couple of problems:

1. I couldn't create a proper `pointerdown` event with `fireEvent.pointerDown` because it doesn't let me specify the `button` event parameter, see https://github.com/testing-library/react-testing-library/issues/838.
2. I couldn't (easily) create a custom `PointerEvent` object because this is not supported in jsdom, see https://github.com/jsdom/jsdom/issues/2527.

Comment (just-boris) [reply src/board/__tests__/board.test.tsx:131]: https://github.com/jsdom/jsdom/issues/2527 also mentions a workaround – using MouseEvent constructor instead

I tried this and it worked, see #236 

user-event library makes it work the same way, but more obscure, because it does a lot more additional side effects

Comment (just-boris) [reply src/board/__tests__/board.test.tsx:131]: This is also a strong signal that we need an official test-util for this scenario (because releasing cursor on `window` is implementation detail).

Can we use #236 for now instead of an extra dependency and I will later open a PR to introduce the official util?

Comment (timogasda) [reply src/board/__tests__/board.test.tsx:131]: Thanks! I swear I tried `MouseEvent("pointerdown")` and it didn't work for me... I must have done something else wrong. Will update this PR.

**Review by just-boris — APPROVED** [review]
